docs/DeveloperGuide.md

```diff
@@ -8,30 +29,194 @@
+perform aggregate operations on the build such as viewing the total price of the build and viewing the total wattage of the build.
+There are also check compatibility functions that check if the build's components are compatible with each other in different ways. Such as:
+- Check if the power supply is compatible with the build's total power consumption
+- Checking if the build's motherboard is compatible with the build's CPU and Cooler
+- Checking if the form factor of the build's case is compatible with the build's motherboard
+- Checking if the build's storage is compatible with the case expansion slots
+- Checking if the number of the build's GPUs is compatible with the motherboard GPU slots
+- Checking if the number of the build's RAM is compatible with the motherboard RAM slots
+
+##### Add & delete component
+
+The user can add and delete components from the build. When the user adds a component, the component is added to the 2D linked hash map. 
+When the user deletes a component, the component is removed from the 2D linked hash map. This is done by calling the `addComponent()` and `deleteComponent()` methods of the Build class.
+Which in turn calls the `addElement()` and `removeElement()` methods of the LinkedHashMap2D class.
+
+![](/images/BuildSequence.png)
+
+### Storage
+
+![](/images/Storage.png)
+
+Note: Some methods are being left out to show the core functionality of the storage class.
+
+Storage is used for loading and saving the user's builds. The user can save their builds to a text file and load them 
+from a text file. 
+
+The `Storage` class has private attributes `FILE_DIRECTORY`, `BUILD_FILE_PATH` and `COMPONENT_FILE_PATH`. 
+`FILE_DIRECTORY` is the directory where the text files are stored. 
+`BUILD_FILE_PATH` is the path to the text file where all the build names are stored. 
+`COMPONENT_FILE_PATH` is the path to the text file where the file is named after the build name. The components of the build 
+are stored in their respective build text file.
+
+The `Storage` class has a constructor that takes in a `BuildManager` object. The `BuildManager` object is used to access the
+builds in the `BuildManager` object. The `Storage` class has a `loadBuild()` method that loads the builds from the text file
+a `saveBuild()` method that saves the builds from the `BuildManager` object into the text file. The `Storage` class also has
+a `loadComponent()` and `saveComponent()` methods utilize the `BuildManager` object to access the `Build` object to load and save
+the components of the build into their respective text files. The `deleteBuild()` method deletes the build from the text file with the path
+`BUILD_FILE_PATH` and deletes the text file with the path `COMPONENT_FILE_PATH` that is named after the build name.
+
+
+### Components
+
+![](/images/Component.png)
+
+Each of the various components are represented by classes stored in the `components` package. The current components represented are the following:
```

Comment:
For this diagram maybe consider using simplified class diagrams and omit the attributes to show an overview or an architecture diagram instead, then include the details later on